core/prisma/schema/schema.prisma

```diff
@@ -451,20 +458,21 @@ enum Action {
 }
 
 model ActionRun {
-  id               String          @id @default(dbgenerated("gen_random_uuid()"))
-  actionInstance   ActionInstance? @relation(fields: [actionInstanceId], references: [id], onDelete: Cascade)
+  id               String            @id @default(dbgenerated("gen_random_uuid()"))
+  actionInstance   ActionInstance?   @relation(fields: [actionInstanceId], references: [id], onDelete: Cascade)
   actionInstanceId String?
-  pub              Pub?            @relation(fields: [pubId], references: [id])
+  pub              Pub?              @relation(fields: [pubId], references: [id])
   pubId            String?
   config           Json?
   event            Event?
   params           Json?
   status           ActionRunStatus
   result           Json
-  user             User?           @relation(fields: [userId], references: [id])
+  user             User?             @relation(fields: [userId], references: [id], onDelete: SetNull)
   userId           String?
-  createdAt        DateTime        @default(now())
-  updatedAt        DateTime        @default(now()) @updatedAt
+  createdAt        DateTime          @default(now())
+  updatedAt        DateTime          @default(now()) @updatedAt
+  PubValueHistory  PubValueHistory[]
```

Comment:
it's kind of annoying all of these tables need a references to `XHistory`, but so be it